### PR TITLE
Add YCbCr Swizzle VUIDs

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-VkImageViewCreateInfo-pNext-01970
VUID-VkSamplerYcbcrConversionCreateInfo-components-02581
VUID-VkSamplerYcbcrConversionCreateInfo-components-02582
VUID-VkSamplerYcbcrConversionCreateInfo-components-02583
VUID-VkSamplerYcbcrConversionCreateInfo-components-02584
VUID-VkSamplerYcbcrConversionCreateInfo-components-02585

### CODE REVIEW

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 1]
```diff
@@ -9485,6 +9485,175 @@ TEST_F(VkLayerTest, CreateYCbCrSampler) {
...
+    ASSERT_NO_FATAL_FAILURE(InitFramework(m_errorMonitor));
+
+    PFN_vkGetPhysicalDeviceFeatures2KHR vkGetPhysicalDeviceFeatures2KHR =
+        (PFN_vkGetPhysicalDeviceFeatures2KHR)vk::GetInstanceProcAddr(instance(), "vkGetPhysicalDeviceFeatures2KHR");
+    ASSERT_TRUE(vkGetPhysicalDeviceFeatures2KHR != nullptr);
+
+    if (DeviceExtensionSupported(gpu(), nullptr, VK_KHR_SAMPLER_YCBCR_CONVERSION_EXTENSION_NAME)) {
+        m_device_extension_names.push_back(VK_KHR_SAMPLER_YCBCR_CONVERSION_EXTENSION_NAME);
+        m_device_extension_names.push_back(VK_KHR_MAINTENANCE1_EXTENSION_NAME);
+        m_device_extension_names.push_back(VK_KHR_BIND_MEMORY_2_EXTENSION_NAME);
```
Comment (jeremy-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:9507]: Do you need all these?

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -9485,6 +9485,175 @@ TEST_F(VkLayerTest, CreateYCbCrSampler) {
...
+    // make sure components.r and components.b can be swapped
+    m_errorMonitor->ExpectSuccess();
+    sycci.components = identity;
+    sycci.components.r = VK_COMPONENT_SWIZZLE_B;
+    sycci.components.b = VK_COMPONENT_SWIZZLE_R;
+    vkCreateSamplerYcbcrConversionFunction(device(), &sycci, NULL, &ycbcr_conv);
+    vkDestroySamplerYcbcrConversionFunction(device(), ycbcr_conv, nullptr);
+    m_errorMonitor->VerifyNotFound();
+
+    // Make sure componenets doesn't affect _444 formats
```
Comment (jeremy-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:9610]: s/componenets/components

File: layers/parameter_validation_utils.cpp [hunk 3]
```diff
@@ -4937,9 +4955,57 @@ bool StatelessValidation::ValidateCreateSamplerYcbcrConversion(VkDevice device,
...
+        if ((components.b != VK_COMPONENT_SWIZZLE_B) && (components.b != VK_COMPONENT_SWIZZLE_IDENTITY) &&
+            (components.b != VK_COMPONENT_SWIZZLE_R)) {
+            skip |= LogError(device, "VUID-VkSamplerYcbcrConversionCreateInfo-components-02584",
+                             "%s: When using a XChroma subsampled format (%s) the components.b needs to be VK_COMPONENT_SWIZZLE_B "
+                             "or VK_COMPONENT_SWIZZLE_IDENTITY or VK_COMPONENT_SWIZZLE_R, but is %s.",
+                             apiName, string_VkFormat(pCreateInfo->format), string_VkComponentSwizzle(components.b));
+        }
+
+        // If one is identity, both need to be
+        bool rIdentity = ((components.r == VK_COMPONENT_SWIZZLE_R) || (components.r == VK_COMPONENT_SWIZZLE_IDENTITY));
```
Comment (jeremy-lunarg) [inline layers/parameter_validation_utils.cpp:4998]: nit: rIdentity and bIdentity can be const

**Comment by jeremy-lunarg** [comment]
Running through CI.

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:9507]: Welcome to [VK_KHR_sampler_ycbcr_conversion](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_KHR_sampler_ycbcr_conversion.html) which requires all of these to be enabled as well

Comment (jeremy-lunarg) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:9507]: wow... you aren't kidding...

**Pull request merged**